### DISCUSSION TITLE
When using $watch, the callback function can't access the current variables

### DISCUSSION THREAD

**Discussion by ismailnakkar** [alpinejs/alpine#1393] [7. Bugs (archived)]
For instance, the console.log in the following prints undefined after changing the value of x:

```
<div x-data="dropdown()" x-init="$watch('x', print)">
    <button @click="x++">Click</button>
    <p x-text="x"></p>
</div>

<script>
    function dropdown() {
        return {
            x: 0,
            print() {
                console.log(this.x)
            }
        }
    }
</script>
```

**Comment by ryangjchandler** [comment]
This isn't an issue with Alpine, rather show JavaScript is binding `this`. 

I'm not at my desk right now so can't actually type out a good example, but you either want to forcefully bind to `this` using `print.bind`, or instead do something like this:

```js
$watch('x', () => print())
```
